packages/dds/tree/src/edit-manager/editManager.ts

```diff
@@ -137,12 +135,32 @@ export class EditManager<TChangeset, TChangeFamily extends ChangeFamily<any, TCh
         return netChange;
     }
 
-    private rebaseBranch(branch: Branch<TChangeset>, newRef: SeqNumber) {
-        const trunkChanges = this.getCommitsBetween(branch.refSeq, newRef);
+    /**
+     * Updates the `branch` to reflect the local changes that the session owner would have had after
+     * they learned of the commit with sequence number `newRef` being sequenced.
+     * Concretely, this means two things:
+     * 1. Rebasing the changes in the branch over any new trunk changes up to and including `newRef`.
+     * 2. Removing from the branch those changes that were sequenced before or at sequence number `newRef`.
+     * We can purge those because any new changes from this session (such as the one whose ref is `newRef`)
+     * will be independent from them (they will instead depend the trunk version of them, or some later changes
+     * in the trunk).
+     * @param branch - The branch to update.
+     * @param newRef - The point in the trunk to rebase the branch up to.
+     */
+    private updateBranch(branch: Branch<TChangeset>, newRef: SeqNumber) {
+        const trunkChanges = this.getCommitsAfterAndUpToInclusive(branch.refSeq, newRef);
+        if (trunkChanges.length === 0) {
+            assert(branch.refSeq === newRef, "Expected trunk changes");
+            // This early return avoids rebasing the branch changes over an empty sandwich.
+            return;
+        }
         const newBranchChanges: Commit<TChangeset>[] = [];
         const inverses: TChangeset[] = [];
 
         for (const commit of branch.localChanges) {
+            // If this commit was sequenced after the ref of the new commit then it means
```

Comment:
I found the wording of this comment somewhat confusing. I also think it's mostly redundant with the comment at the top of the function.

Reply:
Agreed. Removed.